atomic_reactor/plugins/exit_sendmail.py

```diff
@@ -223,6 +169,71 @@ def _send_mail(self, receivers_list, subject, body):
                 s.quit()
 
+    def _get_email_from_koji_obj(self, obj):
+        if obj.get('krb_principal') and obj['krb_principal'] != '':
+            return obj['krb_principal'].lower()
+        else:
+            if not self.email_domain:
+                raise RuntimeError("Empty email_domain specified")
+            return '@'.join([obj['owner_name'], self.email_domain])
+
+    def _get_koji_submitter(self, session, metadata):
+        koji_task_id = metadata['labels']['koji-task-id']
+        koji_task_info = session.getTaskInfo(koji_task_id)
+        koji_task_owner = session.getUser(koji_task_info['owner'])
+        koji_task_owner_email = self._get_email_from_koji_obj(koji_task_owner)
+        self.submitter = koji_task_owner_email
+        return koji_task_owner_email
+
+    def _get_koji_owners(self, session, metadata):
+        result = []
+        koji_build_id = metadata['labels']['koji-build-id']
+        koji_build_info = session.getBuild(koji_build_id)
+
+        koji_tags = session.listTags(koji_build_id)
+        for koji_tag in koji_tags:
+            koji_tag_id = koji_tag['id']
+            koji_package_id = koji_build_info['package_id']
+            koji_pkg_tag_config = session.getPackageConfig(koji_tag_id, koji_package_id)
+            koji_pkg_tag_owner = session.getUser(koji_pkg_tag_config['owner'])
+
+            result.append(self._get_email_from_koji_obj(koji_pkg_tag_owner))
+
+        return result
+
+    def _get_receivers_list(self):
+        receivers_list = []
+        if self.always_notify:
+            receivers_list.append(self.always_notify)
+
+        if self.to_koji_submitter or self.to_koji_pkgowner:
+            try:
+                metadata = get_build_json().get("metadata", {})
+                session = create_koji_session(self.koji_hub, self.koji_auth_info)
+            except Exception as e:
+                self.log.exception("Failed to connect to koji")
+            else:
+                if self.to_koji_submitter:
+                    try:
+                        koji_task_owner_email = self._get_koji_submitter(session, metadata)
+                    except Exception as e:
+                        self.log.exception("Failed to include a task submitter")
+                    else:
+                        receivers_list.append(koji_task_owner_email)
+
+                if self.to_koji_pkgowner:
+                    try:
+                        koji_task_owner_emails = self._get_koji_owners(session, metadata)
+                    except Exception as e:
+                        self.log.exception("Failed to include a package owner")
+                    else:
+                        receivers_list += koji_task_owner_emails
+
+        if not receivers_list:
+            raise RuntimeError("No recepients found")
+
+        return receivers_list
+
     def run(self):
         # verify that given states are subset of allowed states
         unknown_states = set(self.send_on) - self.allowed_states
```

Comment:
I can't comment further than this, but we should use: `self.workflow.build_process_failed` instead of `self.workflow.build_result.is_failed()`

By using `build_process_failed` we'll treat other plugin failures (not just docker build failures) as not success state.